x/staking/keeper/msg_server_test.go

```diff
@@ -707,6 +709,104 @@ func TestValidatorBond(t *testing.T) {
+		ValidatorAddress: validatorAddress.String(),
+		Amount:           stakeToken,
+	}
+	tokenizeMsg := types.MsgTokenizeShares{
+		DelegatorAddress:    delegatorAddress.String(),
+		ValidatorAddress:    validatorAddress.String(),
+		Amount:              stakeToken,
+		TokenizedShareOwner: delegatorAddress.String(),
+	}
+	redeemMsg := types.MsgRedeemTokensforShares{
+		DelegatorAddress: delegatorAddress.String(),
+	}
+	disableMsg := types.MsgDisableTokenizeShares{
+		DelegatorAddress: delegatorAddress.String(),
+	}
+	enableMsg := types.MsgEnableTokenizeShares{
+		DelegatorAddress: delegatorAddress.String(),
+	}
+
+	// Delegate normally
+	_, err := msgServer.Delegate(sdk.WrapSDKContext(ctx), &delegateMsg)
+	require.NoError(t, err, "no error expected when delegating")
+
+	// Tokenize shares - it should succeed
+	_, err = msgServer.TokenizeShares(sdk.WrapSDKContext(ctx), &tokenizeMsg)
+	require.NoError(t, err, "no error expected when tokenizing shares for the first time")
+
+	liquidToken := app.BankKeeper.GetBalance(ctx, delegatorAddress, validatorAddress.String()+"/1")
+	require.Equal(t, stakeAmount.Int64(), liquidToken.Amount.Int64(), "user received token after tokenizing share")
+
+	// Redeem to remove all tokenized shares
+	redeemMsg.Amount = liquidToken
+	_, err = msgServer.RedeemTokens(sdk.WrapSDKContext(ctx), &redeemMsg)
+	require.NoError(t, err, "no error expected when redeeming")
+
+	// Attempt to disable again, now it should succeed
+	_, err = msgServer.DisableTokenizeShares(sdk.WrapSDKContext(ctx), &disableMsg)
+	require.NoError(t, err, "no error expected when disabling tokenization")
+
+	// Disabling again while the lock is already in place, does nothing, but should not error
+	_, err = msgServer.DisableTokenizeShares(sdk.WrapSDKContext(ctx), &disableMsg)
+	require.NoError(t, err, "no error expected when disabling again")
+
+	// Attempt to tokenize, it should fail since tokenization is disabled
+	_, err = msgServer.TokenizeShares(sdk.WrapSDKContext(ctx), &tokenizeMsg)
+	require.ErrorIs(t, err, types.ErrTokenizeSharesDisabledForAccount)
+
+	// Now enable tokenization
+	_, err = msgServer.EnableTokenizeShares(sdk.WrapSDKContext(ctx), &enableMsg)
+	require.NoError(t, err, "no error expected when enabling tokenization")
+
+	// Attempt to tokenize again, it should still fail since the unboning period has
+	// not passed and the lock is still active
+	_, err = msgServer.TokenizeShares(sdk.WrapSDKContext(ctx), &tokenizeMsg)
+	require.ErrorIs(t, err, types.ErrTokenizeSharesDisabledForAccount)
+	require.ErrorContains(t, err, fmt.Sprintf("tokenization will be allowed at %s",
+		blockTime.Add(unbondingPeriod)))
+
+	// Increment the block time by the unbonding period and remove the expired locks
+	ctx = ctx.WithBlockTime(ctx.BlockTime().Add(unbondingPeriod))
+	app.StakingKeeper.RemoveExpiredTokenizeShareLocks(ctx, ctx.BlockTime())
+
+	// Attempt to tokenize again, it should succeed this time since the lock has expired
```

Comment:
feels a bit unnecessary since none of the redemption code was modified by this PR, so would prefer to leave it out. But happy to add it if you feel more strongly about it

Reply:
Makes sense to leave out given none of the code was touched, was being overly conservative